app/api/orders.py

```diff
@@ -313,6 +318,14 @@ def before_update_object(self, order, data, view_kwargs):
             if order.order_notes and data['order_notes'] not in order.order_notes.split(","):
                 data['order_notes'] = '{},{}'.format(order.order_notes, data['order_notes'])
 
+        status = data.get('status', None)
+        if status and ((order.status == 'completed' and status in ['expired', 'initializing', 'pending', 'placed']) or
+            (order.status == 'expired' and status in ['completed', 'initializing', 'placed', 'pending']) or
```

Comment:
I think that this check can be optimised by making it into `if (status and order.status != status)`